tests/helper/test-case.js

```diff
@@ -0,0 +1,122 @@
+"use strict";
+
+var fs = require("fs");
+var assert = require("chai").assert;
+var PrismLoader = require("./prism-loader");
+
+/**
+ * Handles parsing of a test case file.
+ *
+ *
+ * A test case file consists of at least two parts, separated by a line of dashes.
+ * This separation line must start at the beginning of the line and consist of at least three dashes.
+ *
+ * The test case file can either consist of two parts:
+ *
+ *     {source code}
+ *     ----
+ *     {expected token stream}
+ *
+ *
+ * or of three parts:
+ *
+ *     {source code}
+ *     ----
+ *     {expected token stream}
+ *     ----
+ *     {text comment explaining the test case}
+ *
+ * If the file contains more than three parts, the remaining parts are just ignored.
+ * If the file however does not contain at least two parts (so no expected token stream),
+ * the test case will later be marked as failed.
+ *
+ *
+ * @type {{runTestCase: Function, transformCompiledTokenStream: Function, parseTestCaseFile: Function}}
+ */
+module.exports = {
+
+	/**
+	 * Runs the given test case file and asserts the result
+	 *
+	 * The passed language identifier can either be a language like "css" or a composed language
+	 * identifier like "css+markup". Composed identifiers can be used for testing language inclusion.
+	 *
+	 * When testing language inclusion, the first given language is the main language which will be passed
+	 * to Prism for highlighting ("css+markup" will result in a call to Prism to highlight with the "css" grammar).
+	 * But it will be ensured, that the additional passed languages will be loaded too.
+	 *
+	 * The languages will be loaded in the order they were provided.
+	 *
+	 * @param {string} languageIdentifier
+	 * @param {string} filePath
+	 */
+	runTestCase: function (languageIdentifier, filePath) {
+		var testCase = this.parseTestCaseFile(filePath);
+		var languages = languageIdentifier.split("+");
+
+		if (null === testCase) {
+			throw new Error("Test case file has invalid format, please read the docs.");
+		}
+
+		var Prism = PrismLoader.createInstance(languages);
+		// the first language is the main language to highlight
+		var mainLanguageGrammar = Prism.languages[languages[0]];
+		var compiledTokenStream = Prism.tokenize(testCase.testSource, mainLanguageGrammar);
+		var simplifiedTokenStream = this.transformCompiledTokenStream(compiledTokenStream);
+
+		assert.deepEqual(simplifiedTokenStream, testCase.expectedTokenStream, testCase.comment);
+	},
+
+
+	/**
+	 * Simplifies the token stream to ease the matching with the expected token stream
+	 *
+	 * @param {string} tokenStream
+	 * @returns {Array.<string[]>}
+	 */
+	transformCompiledTokenStream: function (tokenStream) {
+		return tokenStream.filter(
+			function (token) {
+				// only support objects
+				return (typeof token === "object");
+			}
+		).map(
+			function (entry)
+			{
+				return [entry.type, entry.content];
```

Comment:
@apfelbox: This should be made recursive, or it will not be able to handle `inside` parts.

Maybe something like:
```javascript
return [
	entry.type,
	(typeof entry.content === "object") ? self.transformCompiledTokenStream(entry.content) : entry.content
];
```
(with `self` being assigned `this`)